File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/duckdb/FixedPointTypes.scala
package dev.typr.foundations.docs.duckdb
import dev.typr.scalafoundations.*
import dev.typr.scalafoundations.data.*

@SuppressWarnings(Array("unused"))
object FixedPointTypes:
  //start
  val decimalType: DuckDbType[BigDecimal] = DuckDbTypes.decimal
  val precise: DuckDbType[BigDecimal] = DuckDbTypes.decimal(18, 6)  // DECIMAL(18,6)
  //stop

File: documentation-examples-kotlin/src/kotlin/dev/typr/foundations/docs/db2/FixedPointTypes.kt
package dev.typr.foundations.docs.db2

import dev.typr.kotlinfoundations.*
import dev.typr.kotlinfoundations.data.*
import java.math.BigDecimal

@Suppress("unused")
class FixedPointTypes {
    //start
    val decType: Db2Type<BigDecimal> = Db2Types.decimal
    val preciseType: Db2Type<BigDecimal> = Db2Types.decimal(10, 2)
    val decfloatType: Db2Type<BigDecimal> = Db2Types.decfloat
    val decfloat16: Db2Type<BigDecimal> = Db2Types.decfloat(16)
    //stop
}
